Installed in HEAD.  Needed for glibc-2_7-branch.


2007-11-13  Ulrich Drepper  <drepper@redhat.com>

	* include/kernel-features.h: Moved to...
	* sysdeps/mach/hurd/kernel-features.h: ...here.

2007-11-11  Roland McGrath  <roland@frob.com>

	* include/kernel-features.h: New file.

diff --git a/sysdeps/mach/hurd/kernel-features.h b/sysdeps/mach/hurd/kernel-features.h
new file mode 100644
index 0000000..ad159aa
--- /dev/null
+++ b/sysdeps/mach/hurd/kernel-features.h
@@ -0,0 +1,31 @@
+/* Set flags signalling availability of certain operating system features.
+   Copyright (C) 2007 Free Software Foundation, Inc.
+   This file is part of the GNU C Library.
+
+   The GNU C Library is free software; you can redistribute it and/or
+   modify it under the terms of the GNU Lesser General Public
+   License as published by the Free Software Foundation; either
+   version 2.1 of the License, or (at your option) any later version.
+
+   The GNU C Library is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   Lesser General Public License for more details.
+
+   You should have received a copy of the GNU Lesser General Public
+   License along with the GNU C Library; if not, write to the Free
+   Software Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA
+   02111-1307 USA.  */
+
+/* This file can define __ASSUME_* macros checked by certain source files.
+   Almost none of these are used outside of sysdeps/unix/sysv/linux code.
+   But those referring to POSIX-level features like O_* flags can be.  */
+
+#include <fcntl.h>
+
+/* If a system defines the O_CLOEXEC constant but it is sometimes ignored,
+   it must override this file to define __ASSUME_O_CLOEXEC conditionally
+   (or not at all) to indicate when O_CLOEXEC actually works.  */
+#ifdef O_CLOEXEC
+# define __ASSUME_O_CLOEXEC	1
+#endif
